### PR TITLE
add fallback behavior

### PR DESCRIPTION
**Pull request by jensens**

### CODE REVIEW

File: src/node/behaviors/fallback.rst [hunk 1]
```diff
@@ -0,0 +1,90 @@
+=========================
+node.behaviors.invalidate
```
Comment (rnixx) [inline src/node/behaviors/fallback.rst]: wrong headline

File: src/node/behaviors/fallback.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    current = node
+    for name in path[len(current.path):]:
+        new_current = current.get(name, _marker)
+        if new_current is _marker:
+            return _to_root(current, path=path, visited=visited)
+        current = new_current
+    return current
+
+
+class Fallback(Behavior):
```
Comment (rnixx) [inline src/node/behaviors/fallback.py]: each other behaviour has a related interface, pls add

File: src/node/behaviors/fallback.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+
+def _to_root(node, path, visited):
+    """go in direction root until node with fallback found.
+    or no more parent (breaks)
+    """
+    parent = node.__parent__
+    if parent is None:
+        return _marker
+    if not getattr(parent, 'fallback_key', _marker) or node in visited:
```
Comment (rnixx) [inline src/node/behaviors/fallback.py]: should be IFallback.providedBy as soon as related interface is added

Comment (jensens) [inline src/node/behaviors/fallback.py]: that would increase the effort (time needed). and its called often. keep this.

File: src/node/behaviors/fallback.py (hunk 2 above)
Comment (jensens) [inline src/node/behaviors/fallback.py]: done

**Comment by rnixx** [comment]
I did some docs polishing. Please read whether there is something left needing explanation, otherwise I'm fine with merging.

Thanks!

**Comment by jensens** [comment]
thx @rnixx , looks much better now

**Pull request merged**